cdap-api-spark/src/main/java/co/cask/cdap/api/spark/JavaSparkMain.java

```diff
@@ -55,12 +57,12 @@
  * </pre></code>
  * </p>
  */
-public abstract class JavaSparkMain {
+public interface JavaSparkMain extends Serializable {
```

Comment:
Why does the main program have to be serializable? I don't think that is the case for Vanilla Spark.